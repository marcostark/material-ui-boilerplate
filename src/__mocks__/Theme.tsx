import React, { ReactElement } from 'react'
import { StylesProvider, StylesOptions } from '@material-ui/styles'
import { ThemeProvider as MuiThemeProvider } from '@material-ui/core/styles'
import { ThemeProvider } from 'styled-components'
import theme from 'styles/theme'

const generateClassName: StylesOptions['generateClassName'] = (rule, sheet) =>
  `${sheet?.options.classNamePrefix ?? 'classname'}-${rule.key}`

type Props = {
  children: ReactElement | ReactElement[]
}

const Theme = ({ children }: Props) => (
  <StylesProvider generateClassName={generateClassName}>
    <MuiThemeProvider theme={theme}>
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
    </MuiThemeProvider>
  </StylesProvider>
)

export default Theme
