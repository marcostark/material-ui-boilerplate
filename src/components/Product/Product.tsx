import React, { memo } from 'react'
import Card from '@material-ui/core/Card'
import CardActionArea from '@material-ui/core/CardActionArea'
import CardMedia from '@material-ui/core/CardMedia'
import CardActions from '@material-ui/core/CardActions'
import CardContent from '@material-ui/core/CardContent'
import Typography from '@material-ui/core/Typography'
import IconButton from '@material-ui/core/IconButton'
import AddShoppingCartIcon from '@material-ui/icons/AddShoppingCart'
import ShareIcon from '@material-ui/icons/Share'

type Props = {
  id: number | string
  image: string
}

const Product = (props: Props) => (
  <Card>
    <CardActionArea>
      <CardMedia
        component="img"
        title="Contemplative Reptile"
        height="140"
        image={props.image}
      />
      <CardContent>
        <Typography gutterBottom variant="h5" component="h2">
          Lizard
        </Typography>
        <Typography variant="body2" color="textSecondary" component="p">
          Lizards are a widespread group of squamate reptiles, with over 6,000
          species, ranging across all continents except Antarctica
        </Typography>
      </CardContent>
    </CardActionArea>
    <CardActions>
      <IconButton
        color={props.id === 1 ? 'secondary' : 'default'}
        aria-label="add to favorites"
      >
        <AddShoppingCartIcon />
      </IconButton>
      <IconButton aria-label="share">
        <ShareIcon />
      </IconButton>
    </CardActions>
  </Card>
)

export default memo(Product)
