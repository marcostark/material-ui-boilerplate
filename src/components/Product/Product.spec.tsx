import React from 'react'
import { Router } from 'react-router-dom'
import { createMemoryHistory, History, State } from 'history'
import { render } from '@testing-library/react'
import Theme from '__mocks__/Theme'
import Product from './Product'

describe('<Product />', () => {
  let history: History<State>

  beforeEach(() => {
    history = createMemoryHistory()
  })

  it('should render correctly', () => {
    const { container } = render(
      <Router history={history}>
        <Theme>
          <Product id={1} image="testing.png" />
        </Theme>
      </Router>
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
