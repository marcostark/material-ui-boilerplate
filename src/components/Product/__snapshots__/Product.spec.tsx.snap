// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Product /> should render correctly 1`] = `
<div
  class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <button
    class="MuiButtonBase-root MuiCardActionArea-root"
    tabindex="0"
    type="button"
  >
    <img
      class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
      height="140"
      src="testing.png"
      title="Contemplative Reptile"
    />
    <div
      class="MuiCardContent-root"
    >
      <h2
        class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
      >
        Lizard
      </h2>
      <p
        class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
      >
        Lizards are a widespread group of squamate reptiles, with over 6,000 species, ranging across all continents except Antarctica
      </p>
    </div>
    <span
      class="MuiCardActionArea-focusHighlight"
    />
    <span
      class="MuiTouchRipple-root"
    />
  </button>
  <div
    class="MuiCardActions-root MuiCardActions-spacing"
  >
    <button
      aria-label="add to favorites"
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11 9h2V6h3V4h-3V1h-2v3H8v2h3v3zm-4 9c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2zm-9.83-3.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.86-7.01L19.42 4h-.01l-1.1 2-2.76 5H8.53l-.13-.27L6.16 6l-.95-2-.94-2H1v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.13 0-.25-.11-.25-.25z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      aria-label="share"
      class="MuiButtonBase-root MuiIconButton-root"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7s-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81 1.66 0 3-1.34 3-3s-1.34-3-3-3-3 1.34-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9c-1.66 0-3 1.34-3 3s1.34 3 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.16c-.05.21-.08.43-.08.65 0 1.61 1.31 2.92 2.92 2.92 1.61 0 2.92-1.31 2.92-2.92s-1.31-2.92-2.92-2.92z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`
